yabgp/handler/default_handler.py

```diff
@@ -97,16 +97,30 @@ def get_last_seq_and_file(msg_path):
         file_list.sort()
         msg_file_name = file_list[-1]
         try:
-            with open(msg_path + msg_file_name, 'r') as fh:
-                line = None
-                for line in fh:
-                    pass
-                last = line
-                if line:
-                    if last.startswith('['):
-                        last_seq = eval(last)[1]
-                    elif last.startswith('{'):
-                        last_seq = json.loads(last)['seq']
+            with open(msg_path + msg_file_name, 'rb') as fh:
+                offset = -500
+                break_flag = False
+                while True:
+                    fh.seek(offset, 2)
+                    lines = fh.readlines()
+                    if len(lines) >= 5:
+                        for i in range(len(lines)-1, 0, -1):
+                            last = lines[i]
+                            try:
+                                if last.startswith('['):
+                                    last_seq = eval(last)[1]
+                                    break_flag = True
+                                    break
+                                elif last.startswith('{'):
+                                    last_seq = json.loads(last)['seq']
+                                    break_flag = True
+                                    break
+                            except Exception as e:
+                                LOG.error(e)
+                                LOG.error(last)
+                    if break_flag:
+                        break
+                    offset *= 2
```

Comment:
Basically, you use try-except to get the last real seq(if last line is complete) or the seq before last line(if last line is incomplete) as last seq.

how about if catch an exception, and we know exactly that exception is caused by last line incomplete, we return the last seq as last seq + 1.

```
{seq: 2}
{se.....}  # incomplete
```
and we return last seq = 3, because the next line should be started from seq =3

how do you think @meidli